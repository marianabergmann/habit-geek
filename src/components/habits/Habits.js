import React from "react";

import NewHabit from "./NewHabit";
import Dashboard from "./Dashboard";

const Habits = ({
  habits,
  username,
  createHabit,
  isCreatingHabit,
  startHabitCreation
}) => {
  return isCreatingHabit ? (
    <NewHabit createHabit={createHabit} />
  ) : (
    <Dashboard
      username={username}
      habits={habits}
      startHabitCreation={startHabitCreation}
    />
  );
};

export default Habits;
