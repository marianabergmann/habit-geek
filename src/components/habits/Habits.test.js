import React from "react";
import { shallow } from "enzyme";
import Habits from "./Habits";

describe("Habits component", () => {
  it("Renders Habits list", () => {
    const habits = shallow(<Habits isCreatingHabit={false} />);
    expect(habits).toMatchSnapshot();
  });

  it("Renders new Habit to create a new one", () => {
    const habits = shallow(<Habits isCreatingHabit={true} />);
    expect(habits).toMatchSnapshot();
  });
});
