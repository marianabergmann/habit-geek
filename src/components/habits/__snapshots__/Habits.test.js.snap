// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Habits component Renders Habits list 1`] = `<Dashboard />`;

exports[`Habits component Renders new Habit to create a new one 1`] = `<NewHabit />`;
